tracing-core/src/metadata.rs

```diff
@@ -273,6 +273,47 @@ impl fmt::Display for Level {
     }
 }
 
+#[derive(Debug)]
+/// Returned if parsing a `Level` fails.
+pub struct ParseLevelError {
+    _p: (),
+}
+
+impl fmt::Display for ParseLevelError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.pad(
+            "error parsing level: expected one of \"error\", \"warn\", \
+             \"info\", \"debug\", \"trace\", or a number 1-5",
+        )
+    }
+}
+
+impl std::error::Error for ParseLevelError {}
+
+impl FromStr for Level {
```

Comment:
Style nit: can we move this so that it's next to the other impls for `Level`, instead of after the impls for the error type?